File: app/src/main/java/ru/mrlargha/stemobile/ui/login/LoginResult.java
package ru.mrlargha.stemobile.ui.login;

import androidx.annotation.Nullable;

import ru.mrlargha.stemobile.data.model.LoginServerReply;

class LoginResult {
    @Nullable
    private LoginServerReply success;
    @Nullable
    private String error;

    LoginResult(@Nullable String error) {
        this.error = error;
    }

    LoginResult(@Nullable LoginServerReply success) {
        this.success = success;
    }

    @Nullable
    LoginServerReply getSuccess() {
        return success;
    }

    @Nullable
    String getError() {
        return error;
    }
}

File: app/src/main/java/ru/mrlargha/stemobile/ui/login/LoginResult.kt
package ru.mrlargha.stemobile.ui.login

import ru.mrlargha.stemobile.data.model.LoginServerReply

internal class LoginResult {
    var success: LoginServerReply? = null
        private set
    var error: String? = null
        private set

    constructor(error: String?) {
        this.error = error
    }

    constructor(success: LoginServerReply?) {
        this.success = success
    }

}